packages/contracts/contracts/L1/ValidatorManager.sol

```diff
@@ -0,0 +1,687 @@
+                block.timestamp,
+            "ValidatorManager: min change seconds of commission rate has not elapsed"
+        );
+
+        require(
+            newCommissionRate <= COMMISSION_RATE_DENOM,
+            "ValidatorManager: the max value of commission rate has been exceeded"
+        );
+
+        uint8 oldCommissionRate = validatorInfo.commissionRate;
+        require(
+            newCommissionRate != oldCommissionRate,
+            "ValidatorManager: cannot change to the same value"
+        );
+
+        uint8 changeRange;
+        if (newCommissionRate > oldCommissionRate) {
+            changeRange = newCommissionRate - oldCommissionRate;
+        } else {
+            changeRange = oldCommissionRate - newCommissionRate;
+        }
+        require(
+            changeRange <= validatorInfo.commissionMaxChangeRate,
+            "ValidatorManager: max change rate of commission rate has been exceeded"
+        );
+
+        validatorInfo.commissionRate = newCommissionRate;
+        validatorInfo.commissionRateChangedAt = uint128(block.timestamp);
+
+        emit ValidatorCommissionRateChanged(msg.sender, oldCommissionRate, newCommissionRate);
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function tryUnjail() external {
+        require(getStatus(msg.sender) == ValidatorStatus.IN_JAIL, "ValidatorManager: not in jail");
+
+        require(
+            _jail[msg.sender] <= block.timestamp,
+            "ValidatorManager: jail period has not elasped"
+        );
+
+        delete _jail[msg.sender];
+
+        _resetNoSubmissionCount(msg.sender);
+
+        emit ValidatorUnjailed(msg.sender);
+    }
+
+    /**
+     * @notice Slash KRO at the vault of the validator.
+     *
+     * @param loser       Address of the loser at the challenge.
+     * @param winner      Address of the winner at the challenge.
+     * @param outputIndex The index of output challenged.
+     */
+    function slash(address loser, address winner, uint256 outputIndex) external onlyColosseum {
+        uint128 amountToSlash = ASSET_MANAGER.modifyBalanceWithSlashing(loser, 0, true);
+        _pendingChallengeReward[outputIndex] = amountToSlash;
+
+        updateValidatorTree(loser, true);
+
+        emit Slashed(loser, winner, amountToSlash);
+    }
+
+    /**
+     * @inheritdoc IValidatorManager
+     */
+    function checkSubmissionEligibility(address validator) external view onlyL2OutputOracle {
+        address _nextValidator = nextValidator();
+
+        if (_nextValidator != Constants.VALIDATOR_PUBLIC_ROUND_ADDRESS) {
+            require(
+                validator == _nextValidator,
+                "ValidatorManager: only the next selected validator can submit output"
+            );
+        }
+
+        require(
+            getStatus(validator) == ValidatorStatus.CAN_SUBMIT_OUTPUT,
+            "ValidatorManager: validator should satisfy the condition to submit output"
+        );
```

Comment:
Shouldn't we add checks for the validator is `TRUSTED_VALIDATOR` here, for the output submission at the `ValidatorPool` termination? Maybe `TRUSTED_VALIDATOR` has no authority to submit output in this logic at that time.

Reply:
Before the `ValidatorPool` is terminated, `TRUSTED_VALIDATOR` can register and start it, then it can submit the first output after the termination. Actually I thought that `TRUSTED_VALIDATOR` is same as the other validators except that it is the only one who can submit the first output in `ValidatorManager`. `TRUSTED_VALIDATOR` also needs to register and self-delegate above `MIN_START_AMOUNT`, and needs to be added to the validator tree.